### PR TITLE
[MNT] Modified `test_base_rocketGPU.py`

### PR DESCRIPTION
**Pull request by AnonymousCodes911**
Refactored the test funcs to use `numpy.testing.assert_array_almost_equal` for more robust testing.
### Reference Issues/PRs

#1248 , #1272

### CODE REVIEW

File: aeon/transformations/collection/convolution_based/_rocket.py [hunk 1]
```diff
@@ -68,10 +68,18 @@ class Rocket(BaseCollectionTransformer):
...
+        num_kernels=10_000,
+        normalise=True,
+        n_jobs=1,
+        random_state=None,
+        normalize=False,
+    ):
         self.num_kernels = num_kernels
         self.normalise = normalise
         self.n_jobs = n_jobs
+        self.normalize = normalize
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/convolution_based/_rocket.py:82]: There is already a normalise parameter available.

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py [hunk 2]
```diff
@@ -1,11 +1,20 @@
 """Unit tests for rocket GPU base functionality."""
 
-__maintainer__ = ["hadifawaz1999"]
-__all__ = ["test_base_rocketGPU_univariate", "test_base_rocketGPU_multivariate"]
+__maintainer__ = ["hadifawaz1999", "AnonymousCodes911"]
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:3]: People may @ you about the contents of the file if you add yourself here, FYI.

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py [hunk 3]
```diff
@@ -117,3 +126,30 @@ def test_base_rocketGPU_multivariate():
 
     # check all ppv values are >= 0
     assert (X_transform[:, 0] >= 0).sum() == len(X)
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies("tensorflow", severity="none"),
+    reason="skip test if required soft dependency not available",
+)
+@pytest.mark.parametrize("random_seed, n_channels", [(42, 1), (42, 3)])
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:135]: The random seed can be static, it does not have to be a parameter.

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py [hunk 4]
```diff
@@ -117,3 +126,30 @@ def test_base_rocketGPU_multivariate():
...
+def test_rocket_cpu_gpu(random_seed, n_channels):
+    """Test consistency between CPU and GPU versions of ROCKET."""
+    n_instances = 100
+    if n_channels == 1:
+        X, _ = make_example_2d_numpy()
+    else:
+        X, _ = make_example_3d_numpy()
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:142]: By default `make_example_3d_numpy` only creates an array with 1 channel, be sure to check the documentation. `make_example_3d_numpy` can be used for both univariate and multivariate.

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py [hunk 5]
```diff
@@ -117,3 +126,30 @@ def test_base_rocketGPU_multivariate():
...
+    X = X[:n_instances]
+
+    num_kernels = 100
+
+    rocket_cpu = Rocket(num_kernels=num_kernels, random_state=random_seed)
+    rocket_cpu.fit(X)
+
+    rocket_gpu = ROCKETGPU(random_state=random_seed)
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:150]: The number of kernels is not set here (i think its `n_filters`, not sure why its named differently)

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py [hunk 6]
```diff
@@ -117,3 +126,30 @@ def test_base_rocketGPU_multivariate():
...
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies("tensorflow", severity="none"),
+    reason="skip test if required soft dependency not available",
+)
+@pytest.mark.parametrize("random_seed, n_channels", [(42, 1), (42, 3)])
+def test_rocket_cpu_gpu(random_seed, n_channels):
+    """Test consistency between CPU and GPU versions of ROCKET."""
+    n_instances = 100
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:138]: The functions are creating arrays smaller than this by default

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
A few comments

File: aeon/transformations/collection/convolution_based/_rocket.py (hunk 1 above)
Comment (AnonymousCodes911) [reply aeon/transformations/collection/convolution_based/_rocket.py:82]: I thought we needed a separate parameter for this test, hence I tweaked the spelling, will correct this.

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py (hunk 5 above)
Comment (AnonymousCodes911) [reply aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py:148]: removed `num_kernels` and changed it to `n_filters`

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py (hunk 3 above)
Comment (AnonymousCodes911) [reply aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py]: removed `random_seed` from the parameters

File: aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py (hunk 5 above)
Comment (AnonymousCodes911) [reply aeon/transformations/collection/convolution_based/rocketGPU/tests/test_base_rocketGPU.py]: I dont know how I missed this! 😅

**Comment by hadifawaz1999** [comment]
I see the issue now, the new test will never be able to pass, because ROCKET implementation in aeon still uses directly the number generation with np.random.normal() and sets the seed globally with np.random.seed. However with ROCKETGPU we started using the [new random number generators that numpy introduced](https://numpy.org/doc/stable/reference/random/generator.html) to avoid deprecations in the future (compare both implementations). I tested and it gived different generations on the same seed.

Given this, do we hold this off until we update the random number generation in rocket and hopefully other models that uses np.random @MatthewMiddlehurst  ?

Would like your input on this @TonyBagnall @baraline

**Comment by MatthewMiddlehurst** [comment]
Hmm that makes sense, good to know. I think what we can do is add a `@pytest.mark.xfail` over the test for now, until we can get around to better unifying the present implementations?

**Comment by hadifawaz1999** [comment]
> `pytest.mark.xfail`

Seems like a good solution for now yes

**Comment by AnonymousCodes911** [comment]
based on this discussion, implementing `@pytest.mark.xfail` is a suitable approach should I implement it? @MatthewMiddlehurst

**Review by MatthewMiddlehurst — APPROVED** [review]
LGTM. We can update the issue to note the above.

**Review by hadifawaz1999 — APPROVED** [review]
LGTM! thanks

**Pull request merged**